src/Horus/CairoSemantics.hs

```diff
@@ -402,30 +468,31 @@ mkApConstraints apEnd insts = do
       ap2 <- getAp pcNext
       fp <- getFp
       getApIncrement fp lInst >>= \case
-        Just apIncrement -> let res = assert (ap1 + apIncrement .== ap2) in res
+        Just apIncrement -> assert (ap1 + apIncrement .== ap2)
         Nothing | not isNewStackframe -> assert (ap1 .< ap2)
         Nothing -> pure ()
   lastAp <- getAp lastPc
   when (isRet lastInst) pop
   fp <- getFp
   getApIncrement fp lastLInst >>= \case
-    Just lastApIncrement -> let res = assert (lastAp + lastApIncrement .== apEnd) in res
```

Comment:
How did this weird pattern end up getting several times into the code base? 0_0

Reply:
I use this pattern to do: `let res = huge_expr in trace huge_expr res` and then I removed the trace and failed to fix the code :).